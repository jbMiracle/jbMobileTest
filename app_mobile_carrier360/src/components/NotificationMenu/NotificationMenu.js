import React from 'react';
import { Text, View, FlatList, TouchableOpacity } from 'react-native';
import { Right, Badge } from 'native-base';
import { Menu, MenuOptions, MenuOption, MenuTrigger, renderers } from 'react-native-popup-menu';
import { menuOptionsStyle } from '../C360Header/C360Header.style';
import C360Colors from '../../themes/360Colors';
import C360Icon from '../C360Icon/C360Icon';

const { Popover } = renderers;

const NotificationMenu = ({ notifications, style }) => {
  const {
    headerIcon,
    itemSeparatorStyle,
    menuTriggerStyles,
    popoverProps,
  } = style;
  const notify = menuOptionsStyle({ type: 'notifications' });

  return (
    <Menu renderer={ Popover } rendererProps={ popoverProps }>
      <MenuTrigger customStyles={ menuTriggerStyles }>
        <C360Icon style={ headerIcon } name='jbh-Notification' size={ 23 } color={ C360Colors.White } />
      </MenuTrigger>
      <MenuOptions customStyles={ notify }>
        <FlatList
          data={ truncateNotificationList(notifications) }
          bounces={ false }
          renderItem={ renderNotification(style) }
          ItemSeparatorComponent={ () => <View style={ itemSeparatorStyle } /> }
        />
        { renderViewMore(notifications, style) }
      </MenuOptions>
    </Menu>
  );
};

function renderNotification(style) {
  const notify = menuOptionsStyle({ type: 'notifications' });
  const {
    notificationRow,
    badgeStyle,
    notificationTextView,
    alignFlexEnd,
    notificationHierarchy,
    timeStamp,
    fontSize10,
  } = style;

  return ({ item }) => (
    <MenuOption>
      <View style={ notificationRow }>
        <Badge style={ badgeStyle } />
        <View style={ notificationTextView }>
          <Text style={ [notify.optionText, notificationHierarchy] }> { item.key } </Text>
          <Text style={ notify.optionText }> { item.status } </Text>
        </View>
        <Right style={ alignFlexEnd } >
          <View style={ timeStamp }>
            <C360Icon name='jbh-Circle_Clock' color={ C360Colors.Grey600 } />
            <Text numberOfLines={ 1 } style={ [notify.optionText, fontSize10] }> { item.time } </Text>
          </View>
        </Right>
      </View>
    </MenuOption>
  );
}

function truncateNotificationList(notifications) {
  if (notifications.length > 4) {
    return notifications.slice(0, 4);
  }

  return notifications;
}

function renderViewMore(notifications, style) {
  if (notifications.length < 5) {
    return null;
  }

  return (
    <View style={ style.alignCenter }>
      <TouchableOpacity style={ style.moreNotificationButtonStyle }>
        <Text style={ style.moreNotificationTextStyle }>
          View all Notifications ({ notifications.length })
        </Text>
      </TouchableOpacity>
    </View>
  );
}

export default NotificationMenu;
