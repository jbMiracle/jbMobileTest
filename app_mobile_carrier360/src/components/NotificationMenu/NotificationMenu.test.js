import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NotificationMenu from './NotificationMenu';

it('can render a navigation menu component with a view more button', () => {
  const renderer = new ShallowRenderer();

  const props = {
    notifications: [
      { key: 'Load HV51606 - Status Update', status: 'You have been outbid', time: '2m' },
      { key: 'Favorite Lanes - Available Load', status: 'Dallas, TX, 73019 - San Antonio, TX, 78719', time: '23m' },
      { key: 'Manage Users - Approval Needed', status: 'New User Paul', time: '2d' },
      { key: 'Load HV51607 - Status Update', status: 'You have been outbid', time: '2m' },
      { key: 'Load HV51608 - Status Update', status: 'You have been served', time: '5m' },
    ],
    style: {
      headerIcon: {},
    },
    popoverProps: {},
    menuOptionsStyle: {},
  };

  renderer.render(<NotificationMenu { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('can render a navigation menu component without a view more button', () => {
  const renderer = new ShallowRenderer();

  const props = {
    notifications: [
      { key: 'Load HV51606 - Status Update', status: 'You have been outbid', time: '2m' },
      { key: 'Favorite Lanes - Available Load', status: 'Dallas, TX, 73019 - San Antonio, TX, 78719', time: '23m' },
    ],
    style: {
      headerIcon: {},
    },
    popoverProps: {},
    menuOptionsStyle: {},
  };

  renderer.render(<NotificationMenu { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
