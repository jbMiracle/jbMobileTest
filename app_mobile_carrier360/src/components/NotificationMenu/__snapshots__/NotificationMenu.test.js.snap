// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a navigation menu component with a view more button 1`] = `
<Menu
  onBackdropPress={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  onSelect={[Function]}
  renderer={[Function]}
  rendererProps={Object {}}
>
  <MenuTrigger
    customStyles={Object {}}
    disabled={false}
  >
    <Icon
      allowFontScaling={false}
      color="#FFFFFF"
      name="jbh-Notification"
      size={23}
      style={Object {}}
    />
  </MenuTrigger>
  <MenuOptions
    customStyles={
      Object {
        "optionText": Object {
          "color": "#444444",
          "fontSize": 12,
        },
        "optionTouchable": Object {
          "activeOpacity": 0.3,
          "underlayColor": "transparent",
        },
        "optionWrapper": Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "margin": 5,
        },
        "optionsContainer": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
        },
        "optionsWrapper": Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
          "paddingHorizontal": 3,
          "paddingVertical": 10,
          "width": 726.0406582768636,
        },
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      bounces={false}
      data={
        Array [
          Object {
            "key": "Load HV51606 - Status Update",
            "status": "You have been outbid",
            "time": "2m",
          },
          Object {
            "key": "Favorite Lanes - Available Load",
            "status": "Dallas, TX, 73019 - San Antonio, TX, 78719",
            "time": "23m",
          },
          Object {
            "key": "Manage Users - Approval Needed",
            "status": "New User Paul",
            "time": "2d",
          },
          Object {
            "key": "Load HV51607 - Status Update",
            "status": "You have been outbid",
            "time": "2m",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <View
      style={undefined}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          View all Notifications (
          5
          )
        </Text>
      </TouchableOpacity>
    </View>
  </MenuOptions>
</Menu>
`;

exports[`can render a navigation menu component without a view more button 1`] = `
<Menu
  onBackdropPress={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  onSelect={[Function]}
  renderer={[Function]}
  rendererProps={Object {}}
>
  <MenuTrigger
    customStyles={Object {}}
    disabled={false}
  >
    <Icon
      allowFontScaling={false}
      color="#FFFFFF"
      name="jbh-Notification"
      size={23}
      style={Object {}}
    />
  </MenuTrigger>
  <MenuOptions
    customStyles={
      Object {
        "optionText": Object {
          "color": "#444444",
          "fontSize": 12,
        },
        "optionTouchable": Object {
          "activeOpacity": 0.3,
          "underlayColor": "transparent",
        },
        "optionWrapper": Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "margin": 5,
        },
        "optionsContainer": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
        },
        "optionsWrapper": Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
          "paddingHorizontal": 3,
          "paddingVertical": 10,
          "width": 726.0406582768636,
        },
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      bounces={false}
      data={
        Array [
          Object {
            "key": "Load HV51606 - Status Update",
            "status": "You have been outbid",
            "time": "2m",
          },
          Object {
            "key": "Favorite Lanes - Available Load",
            "status": "Dallas, TX, 73019 - San Antonio, TX, 78719",
            "time": "23m",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </MenuOptions>
</Menu>
`;
